src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,81 @@
+package duke;
+
+import duke.command.Parser;
+import duke.exception.DukeException;
+import duke.exception.DukeIoException;
+import duke.task.TaskList;
+import duke.ui.Ui;
+import javafx.application.Application;
+import javafx.application.Platform;
+import javafx.stage.Stage;
+
+/**
+ * Represents an instance of the Duke application.
+ */
+public class Duke extends Application {
+    /** Global task list for all operations. */
+    private TaskList taskList;
+
+    @Override
+    public void init() throws Exception {
+        super.init();
+        this.initializeTaskList();
+    }
+
+    /**
+     * Performs pre-execution initialization of resources required throughout the application lifecycle.
+     * Loads any existing database and attaches observers for saving the database to disk.
+     * @return The current instance of the application being initialized.
+     */
```

Comment:
```suggestion
    /**
     * Performs pre-execution initialization of resources required throughout the application lifecycle.
     * Loads any existing database and attaches observers for saving the database to disk.
     *
     * @return The current instance of the application being initialized.
     */
```
Good job on all the Javadoc! For this, I think there should be a line break between the description field and return field.

Reply:
Thanks for pointing that out. Fixed in commit 734bff7.